python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1355,40 +1357,57 @@ def can_run_with_parent(
-            or (
-                # there is a backfill policy for the parent
-                parent_node.backfill_policy is not None
-                # the same subset of parents is targeted as the child
-                and parent_target_subset.value == candidate_target_subset.value
-                and (
-                    # there is no limit on the size of a single run or...
-                    parent_node.backfill_policy.max_partitions_per_run is None
-                    # a single run can materialize all requested parent partitions
-                    or parent_node.backfill_policy.max_partitions_per_run
-                    > len(asset_partitions_to_request_map[parent.asset_key])
-                )
-                # all targeted parents are being requested this tick
-                and len(asset_partitions_to_request_map[parent.asset_key])
-                == parent_target_subset.size
+        return False
+    if parent_node.priority_repository_handle is not candidate_node.priority_repository_handle:
+        logger.info(
+            "Excluding {candidate_node.asset_key} from request list. Reason: parent {parent_node.asset_key} and {candidate_node.asset_key} are in different code locations."
+        )
+        return False
+    if (
+        parent.partition_key not in asset_partitions_to_request_map[parent.asset_key]
+        and parent not in candidates_unit
+    ):
+        logger.info(
+            f"Excluding {candidate_node.asset_key} from request list. Reason: parent asset {parent.asset_key} with partition key {parent.partition_key} is not requested in this iteration."
+        )
+        return False
+    if (
+        # if there is a simple mapping between the parent and the child, then
+        # with the parent
+        has_identity_partition_mapping
+        # if there is not a simple mapping, we can only materialize this asset with its
+        # parent if...
+        or (
+            # there is a backfill policy for the parent
+            parent_node.backfill_policy is not None
+            # the same subset of parents is targeted as the child
+            and parent_target_subset.value == candidate_target_subset.value
+            and (
+                # there is no limit on the size of a single run or...
+                parent_node.backfill_policy.max_partitions_per_run is None
+                # a single run can materialize all requested parent partitions
+                or parent_node.backfill_policy.max_partitions_per_run
+                > len(asset_partitions_to_request_map[parent.asset_key])
             )
-            # if all the above are true, then a single run can be launched this tick which
-            # will materialize all requested partitions
+            # all targeted parents are being requested this tick
+            and len(asset_partitions_to_request_map[parent.asset_key]) == parent_target_subset.size
         )
-    )
+    ):
+        return True
+    else:
+        logger.log(
```

Comment:
I had a harder time splitting this last condition apart and that resulted in a much more complicated log string. Could be a candidate for another pass at simplifying